playground/src/lib/api/api.ts

```diff
@@ -0,0 +1,180 @@
+import type { ApiResponse, Workflow } from "@/types";
+import {
+  type TraceDetailSpansResponse,
+  TraceListResponseSchema,
+  TraceSummarySchema,
+} from "@fiberplane/fpx-types";
+import { FetchOpenApiSpecError, isFailedToFetchError } from "./errors";
+import { baseFetch, fpFetch, getFpApiBasePath } from "./fetch";
+import { safeParseBodyText } from "./utils";
+
+export const api = {
+  getWorkflows: async (): Promise<ApiResponse<Workflow[]>> => {
+    return fpFetch<ApiResponse<Workflow[]>>("/api/workflows");
+  },
+
+  getWorkflow: async (id: string): Promise<ApiResponse<Workflow>> => {
+    return fpFetch<ApiResponse<Workflow>>(`/api/workflows/${id}`);
+  },
+
+  createWorkflow: async (data: {
+    prompt: string;
+    openApiSchema: string;
+    summary?: string;
+    description?: string;
+  }): Promise<ApiResponse<Workflow>> => {
+    return fpFetch<ApiResponse<Workflow>>("/api/workflows/create", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(data),
+    });
+  },
+
+  updateWorkflow: async (
+    id: string,
+    data: {
+      prompt: string;
+      openApiSchemaId: string;
+      summary?: string;
+      description?: string;
+    },
+  ): Promise<ApiResponse<Workflow>> => {
+    return fpFetch<ApiResponse<Workflow>>(`/api/workflows/${id}`, {
```

Comment:
`fpFetch` does a bunch of magic that was repeated across all the functions in the `api` object

- throw `FpApiError` errors with additional context that the frontend can use in error screens
- default parse response as json